src/Kernel/ClassDescription.class.st

```diff
@@ -373,186 +371,149 @@ ClassDescription >> commentStamp: changeStamp [
 
-	| method selector |
+	| method |
 
 	method := self compiler
-		source: text;
+		source: code;
 		requestor: requestor;
-		failBlock:  [ ^nil ];
+		failBlock:  [ ^ nil ];
 		compile.
 
-	selector := method selector.
 	logSource ifTrue: [
 		self
-			logMethodSource: (requestor ifNotNil: [ :r | r text ] ifNil: [ text ]) "the requestor text might have been changed by the compiler and may be different thant text argument"
+			logMethodSource: (requestor ifNotNil: [ :r | r text ] ifNil: [ code ]) "the requestor text might have been changed by the compiler and may be different thant text argument"
 			forMethod: method
-			inCategory: category
-			withStamp: changeStamp].
+			inCategory: protocolName
+			withStamp: changeStamp ].
 
-	self
-		addAndClassifySelector: selector
-		withMethod: method
-		inProtocol: category.
+	self addAndClassifySelector: method selector withMethod: method inProtocol: protocolName.
 
-	self instanceSide
-		noteCompilationOfMethod: method
-		meta: self isClassSide.
+	self instanceSide noteCompilationOf: method meta: self isClassSide.
 
-	^ selector
+	^ method selector
 ]
 
 { #category : #compiling }
 ClassDescription >> compile: code notifying: requestor [
 	"Refer to the comment in Behavior|compile:notifying:."
 
-	^self compile: code
-		 classified: Protocol unclassified
-		 notifying: requestor
+	^ self compile: code classified: Protocol unclassified notifying: requestor
 ]
 
 { #category : #compiling }
-ClassDescription >> compileSilently: code classified: category [
-	"Compile the code and classify the resulting method in the given category, leaving no trail in the system log, nor in any change set, nor in the 'recent submissions' list. This should only be used when you know for sure that the compilation will succeed."
+ClassDescription >> compileSilently: code classified: protocolName [
+	"Compile the code and classify the resulting method in the given protocol, leaving no trail in the system log, nor in any change set, nor in the 'recent submissions' list. This should only be used when you know for sure that the compilation will succeed."
 
-	^ self compileSilently: code classified: category notifying: nil
+	^ self compileSilently: code classified: protocolName notifying: nil
 ]
 
 { #category : #compiling }
-ClassDescription >> compileSilently: code classified: category notifying: requestor [
-	"Compile the code and classify the resulting method in the given category, leaving no trail in the system log, nor in any change set, nor in the 'recent submissions' list. This should only be used when you know for sure that the compilation will succeed."
+ClassDescription >> compileSilently: code classified: protocolName notifying: requestor [
+	"Compile the code and classify the resulting method in the given protocol, leaving no trail in the system log, nor in any change set, nor in the 'recent submissions' list. This should only be used when you know for sure that the compilation will succeed."
```

Comment:
The original comment makes no sense. The previsible success of a compilation is not related to leaving a trace.